hi-ml/src/health_ml/utils/logging.py

```diff
@@ -129,9 +140,14 @@ def version(self) -> int:
         return 0
 
     def finalize(self, status: str) -> None:
-        if self.run is not None and self.has_custom_run:
-            # Run.complete should only be called if we created an AzureML run here in the constructor.
-            self.run.complete()
+        if self.enable_logging_outside_azure_ml and not self.is_running_in_azure_ml and self.run is not None:
+            if self.has_user_provided_run:
+                # The logger uses a run that was provided by the user: Flush it, but do not complete it.
```

Comment:
Why do we flush if the user provided a run and complete otherwise?

Reply:
Is this because we don't want to complete the run after training if we still want to use it in inference?

Reply:
Correct, yes. I added a comment.